Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -136,11 +167,58 @@ Vector<StyledMarkedText> StyledMarkedText::subdivideAndResolve(const Vector<Mark
         return { styledMarkedText };
     }
 
-    auto markedTexts = MarkedText::subdivide(textsToSubdivide);
+    auto markedTexts = MarkedText::subdivide(textsToSubdivide, OverlapStrategy::None);
     ASSERT(!markedTexts.isEmpty());
     if (UNLIKELY(markedTexts.isEmpty()))
         return { };
 
+    bool isHighlights = false;
+
+    if (!markedTexts.isEmpty()) {
+        // Get rid of duplicates in vector, check if contains highlights
+        ListHashSet<MarkedText> noDuplicates;
+        for (auto& element : markedTexts) {
+            noDuplicates.add(element);
+            if (!isHighlights && element.type == 6)
+                isHighlights = true;
+        }
+        // Add back to markedTexts
+        if (!noDuplicates.isEmpty()) {
+            markedTexts.clear();
+            for (auto& element : noDuplicates)
+                markedTexts.append(element);
+        }
+        // Sort so highlights follow correct priority/insertion order
+        if (isHighlights) {
+            // Maintain copy of old unordered markedTexts
+            Vector<MarkedText> unorderedMarkedTexts = markedTexts;
+            markedTexts.clear();
+            // Append all highlights first, in order
+            for (auto& highlightName : markedTextsNames) {
+                for (auto& element : unorderedMarkedTexts) {
+                    if (element.highlightName == highlightName)
+                        markedTexts.append(element);
+                }
+            }
+            // Append selection last, to maintain order
+            for (auto& element : unorderedMarkedTexts) {
+                if (element.highlightName.isNull())
+                    markedTexts.append(element);
+            }
+        }
+    }
+
+    if (isHighlights) {
+        frontmostMarkedTexts.reserveInitialCapacity(markedTexts.size());
+        frontmostMarkedTexts.uncheckedAppend(resolveStyleForMarkedText(markedTexts[0], baseStyle, renderer, lineStyle, paintInfo));
+        for (auto it = markedTexts.begin() + 1, end = markedTexts.end(); it != end; ++it)
+            frontmostMarkedTexts.uncheckedAppend(resolveStyleForMarkedText(*it, baseStyle, renderer, lineStyle, paintInfo));
```

Comment:
Hmm.. it isn't obvious to me why the first marked text is treated differently?

Reply:
This looks like we could get away with

```suggestion
forEach(markedTexts, [&](auto& markedText) {
    frontmostMarkedTexts.uncheckedAppend(resolveStyleForMarkedText(*it, baseStyle, renderer, lineStyle, paintInfo));
});
```

Reply:
Yes that's correct. It's not needed anymore. Before we would compare stuff while looping but now we are just appending everything so we do not need to have an element in the vector before looping.